enhancements/network/service-egress-traffic-steering.md

```diff
@@ -0,0 +1,286 @@
+
+The egress part will be handled by OVN-Kubernetes, which will choose a node that acts as the point of ingress/egress, and steer the relevant pods' egress traffic to its mgmt port, by using logical router policies on the ovn_cluster_router.
+When that traffic reaches the node's mgmt port it will use its routing table and iptables.
+Because of that, it will also take care of adding the necessary iptables rules on the selected node to SNAT traffic exiting from these pods to the service's external IP. The node will also be labeled with `egress-service.k8s.ovn.org/<svc-namespace>-<svc-name>: ""`, which can be consumed by a LoadBalancer provider to handle the ingress part.
+
+The ingress part will be handled by a LoadBalancer provider, such as MetalLB, that would need to select the right node (and only it) for announcing the LoadBalancer service (ingress traffic) according to the `egress-service.k8s.ovn.org/<svc-namespace>-<svc-name>: ""` label set by OVN-Kubernetes.
+Taking MetalLB as an example for a LoadBalancer provider, the user will need to create their `L2Advertisement` and/or `BGPAdvertisements` with the `nodeSelectors` field pointing to that label. That way only the node holding the label will be used for announcing the LoadBalancer service IP.
+It is worth noting that in MetalLB's case, a given LoadBalancer service can be announced by multiple L2 and BGP advertisements, possibly being (even accidently) announced from multiple nodes. For our use-case the user MUST take care to configure their MetalLB resources in a way that the service is announced only by the node holding the label - a full example is detailed in [Usage Example](#Usage-Example).
+
+To achieve these goals, we introduce a new annotation for users to set on LoadBalancer services: `k8s.ovn.org/egress-service`, which can be either empty or contain a `network-hint`: `'{"network-hint":{"cidr":"172.19.0.0/24"}}'` with a CIDR.
+By specifying a `network-hint`, only a node that has an interface in that CIDR can be selected for the service's traffic (as explained earlier), and the service's pods egress traffic will exit through that interface.
+By not specifying a `network-hint` any node in the cluster can be chosen to manage the service's traffic, and the traffic will exit according to the node's routing table.
+
+### Implementation Details/Notes/Constraints
+
+A new annotation `k8s.ovn.org/egress-service` is supported for LoadBalancer services.
+When `ovnkube-master` detects that a LoadBalancer service has this annotation it will elect a node to act as the point for all of the traffic of that service (ingress/egress). If the annotation contains a valid CIDR in its `network-hint` only a node that has an interface in that CIDR can be elected, which is possible by inspecting all of the nodes' `k8s.ovn.org/host-addresses` annotation.
```

Comment:
If it forces, I wouldn't call it a `-hint`

Reply:
changed to `interface` instead of `network-hint`